### PR TITLE
Fix Postgres build after rebase on master

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java [hunk 1]
```diff
@@ -76,16 +76,4 @@ public void deleteShouldReturnFalseWhenRowDoesNotExist() {
     public UpdatableTickingClock clock() {
         return clock;
     }
-
-    @Disabled("Delete method always return true (to avoid LWT)")
-    @Override
-    public void deleteShouldReturnTrueWhenRowExists() {
```
Comment (vttranlina) [inline server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java:82]: no, please keep it

File: server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala [hunk 2]
```diff
@@ -217,18 +218,4 @@
      assertThat(SMono.fromPublisher(testee.retrieve(uploadId2, USER)).block())
        .isNotNull
    }
-
-   @Test
-   def deleteShouldReturnTrueWhenRowExists(): Unit = {
```
Comment (vttranlina) [inline server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala:222]: no, please keep it

**Review by quantranhong1999 — APPROVED** [review]

File: server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java (hunk 1 above)
Comment (quantranhong1999) [reply server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java:82]: I doubted this also. But I guess that we removed that test in master?

**Review by hungphan227 — APPROVED** [review]

File: server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java (hunk 1 above)
Comment (vttranlina) [reply server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java:82]: No, it still exist in master branch
https://github.com/apache/james-project/blob/master/server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java#L64

File: server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala (hunk 2 above)
Comment (vttranlina) [reply server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala:222]: https://github.com/apache/james-project/blob/master/server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala#L192

Comment (Arsnael) [reply server/data/data-jmap/src/test/java/org/apache/james/jmap/api/upload/UploadRepositoryContract.scala:222]: It is already there, messed up during the rebase, it was duplicated, thus not compiling. That's just removing the duplicate tests

File: server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java (hunk 1 above)
Comment (Arsnael) [reply server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/upload/CassandraUploadRepositoryTest.java:82]: It is already there, messed up during the rebase, it was duplicated, thus not compiling. That's just removing the duplicate tests

**Comment by vttranlina** [comment]
> https://ci-builds.apache.org/job/james/job/ApacheJames/job/PR-2135/3/
> 
> @vttranlina can you have a look into it tomorrow please? I don't know if it's related to your latest commits after rebase on those classes or not

Can you cherry-pick this commit: https://github.com/apache/james-project/commit/8474663819e1ad3c8569ae931355c7369edfac58

// 
reason: the `Instant` of java21 has more precision than java11 (9 vs 6)
-> How to fix? 
After insert, the uploadDate should return from postgres database

**Comment by Arsnael** [comment]
Cheers @vttranlina , thanks a lot :)

File: server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/upload/PostgresUploadDAO.java [hunk 3]
```diff
@@ -71,14 +71,21 @@ public PostgresUploadDAO(@Named(PostgresExecutor.NON_RLS_INJECT) PostgresExecuto
...
+                .set(PostgresUploadTable.SIZE, upload.sizeAsLong())
+                .set(PostgresUploadTable.BLOB_ID, upload.blobId().asString())
+                .set(PostgresUploadTable.USER_NAME, user.asString())
+                .set(PostgresUploadTable.UPLOAD_DATE, INSTANT_TO_LOCAL_DATE_TIME.apply(upload.uploadDate()))
+                .returning(PostgresUploadTable.ID,
+                    PostgresUploadTable.CONTENT_TYPE,
+                    PostgresUploadTable.SIZE,
+                    PostgresUploadTable.BLOB_ID,
+                    PostgresUploadTable.UPLOAD_DATE,
+                    PostgresUploadTable.USER_NAME)))
```
Comment (quantranhong1999) [inline server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/upload/PostgresUploadDAO.java:87]: My concern about this: Should we pay the overhead of transferring more data just to pass test?

Are there any other possible ways like generating an Instant with second precision, or returning at least just the `UPLOAD_DATE`?

Comment (vttranlina) [reply server/data/data-jmap-postgres/src/main/java/org/apache/james/jmap/postgres/upload/PostgresUploadDAO.java:87]: I thought it first when implemented, 
but only return upload_date, the code looks quite confuse after long time.

**Comment by vttranlina** [comment]
I'm fixing why ci fail
emailGetShouldReturnUncheckedMailAddressValueWhenDraftEmail(GuiceJamesServer) – org.apache.james.jmap.rfc8621.postgres.PostgresEmailSetMethodTest

**Comment by Arsnael** [comment]
> I'm fixing why ci fail emailGetShouldReturnUncheckedMailAddressValueWhenDraftEmail(GuiceJamesServer) – org.apache.james.jmap.rfc8621.postgres.PostgresEmailSetMethodTest

That one was easy to fix.

The other one `PostgresIdentityGetTest.mayDeleteShouldReturnFalseWhenUpdateServerSetIdentity` is trickier, disabled for now

Result of writing new jmap IT and only running those new tests against memory and not distributed... careful on that next time please :)

**Comment by vttranlina** [comment]
> The other one `PostgresIdentityGetTest.mayDeleteShouldReturnFalseWhenUpdateServerSetIdentity` is trickier, disabled for now

Fixed it: https://github.com/apache/james-project/pull/2143

**Review by quantranhong1999 — APPROVED** [review]

**Review by vttranlina — APPROVED** [review]

**Pull request merged**